crates/rapier3d-meshloader/src/lib.rs

```diff
@@ -0,0 +1,88 @@
+//! ## Mesh loader for the Rapier physics engine
+//!
+//! See documentation from [`mesh_loader`] for supported formats.
+//!
+//! Rapier is a set of 2D and 3D physics engines for games, animation, and robotics. The `rapier3d-meshloader`
+//! crate lets you create a shape compatible with `rapier3d` and `parry3d` (the underlying collision-detection
+//! library) from different file formats, see the following features list:
+//! `stl`: support .stl files
+//! `collada`: support .dae files
+//! `wavefront`: support .obj files
+
+#![deny(missing_docs)]
+
+use mesh_loader::Mesh;
+use rapier3d::geometry::{MeshConverter, SharedShape};
+use rapier3d::math::{Isometry, Point, Real};
+use rapier3d::prelude::MeshConverterError;
+use std::path::Path;
+
+/// The result of loading a shape.
+pub struct LoadedShape {
+    /// The shape loaded from the file and converted by the [`MeshConverter`].
+    pub shape: SharedShape,
+    /// The shape’s pose.
+    pub pose: Isometry<Real>,
+    /// The raw mesh read from the file without any modification.
+    pub raw_mesh: Mesh,
+}
+
+/// Error while loading an STL file.
+#[derive(thiserror::Error, Debug)]
+pub enum MeshLoaderError {
+    /// An error triggered by rapier’s [`MeshConverter`].
+    #[error(transparent)]
+    MeshConverter(#[from] MeshConverterError),
+    /// A generic IO error.
+    #[error(transparent)]
+    Io(#[from] std::io::Error),
+}
+
+/// Loads an file as shapes from a file.
+///
+/// # Parameters
+/// - `path`: the file’s path.
+/// - `converter`: controls how the shapes are computed from the content. In particular, it lets
+///                you specify if the computed [`SharedShape`] is a triangle mesh, its convex hull,
+///                bounding box, etc.
+// TODO: call a function for each mesh to load ? To be able to have a different mesh converter?
```

Comment:
We could support this in a new function, yes. The closure should be given whatever metadata is associated to the mesh to help guide their choice of converter. This can be added on a different PR.

Reply:
created https://github.com/dimforge/rapier/issues/756